lib/functions.inc.php

```diff
@@ -833,3 +833,36 @@ function allowed_rate($login,$ip_addr,$rrl_config) {
     }
     return true;
 }
+
+class LtbAttributeValue {
```

Comment:
Maybe it would be better to create a separate file for this class, to lighten the current file

Reply:
what filename do you suggest ?
should it include  ltb_ldap_get_first_available_value as well ?

Reply:
I propose: LtbAttributeValue_class.php

For ltb_ldap_get_first_available_value, good question. I'm not a object oriented programming expert, but I think the class should only contains methods related to the object. So with this rule, the function should not be in the class. What do you think?

Reply:
Having all within same file is better ihmo, and it can then be included only if needed.
To respect OOP then it can be made static method of LtbAttributeValue so
 ltb_ldap_get_first_available_value,  would be : LtbAttributeValue.ldap_get_first_value($ldap, $entry, $attributes)
and use a __construct to be able to call new LtbAttributeValue($attribute,$value) instead of getLtbAttributeValue

Reply:
Seems ok for me